recipes/halide/build.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# On OSX, we need to ensure we're using conda's gcc/g++
+if [[ `uname` == Darwin ]]; then
+     export CC=gcc
+     export LD=gcc
+     export CXX=g++
```

Comment:
Does Halide use OpenMP? From this [search]( https://github.com/halide/Halide/search?utf8=%E2%9C%93&q=omp ), it looks like it generates code that does, but it itself doesn't. As such, I don't think we need the `gcc` package. So, try to follow this PR ( https://github.com/conda-forge/thrift-cpp-feedstock/pull/4 ) as an example.